test/manipulation.test.js

```diff
@@ -2620,22 +2620,43 @@ describe('manipulation', function() {
         });
     });
 
-    it('fails the upsertWithWhere operation when multiple instances are ' +
-        'retrieved based on the filter criteria', function(done) {
-      Person.create([
-        {id: '2', name: 'Howie', city: 'Florida'},
-        {id: '3', name: 'Kevin', city: 'Florida'},
-      ], function(err, instance) {
-        if (err) return done(err);
-        Person.upsertWithWhere({city: 'Florida'}, {
-          id: '4', name: 'Brian',
-        }, function(err) {
-          err.message.should.equal('There are multiple instances found.' +
+    bdd.itIf(connectorCapabilities.atomicUpsertWithWhere !== true,
+      'fails the upsertWithWhere operation when multiple instances are ' +
+      'retrieved based on the filter criteria', function(done) {
+        Person.create([
+          {id: '2', name: 'Howie', city: 'Florida'},
+          {id: '3', name: 'Kevin', city: 'Florida'},
+        ], function(err, instance) {
+          if (err) return done(err);
+          Person.upsertWithWhere({city: 'Florida'}, {
+            id: '4', name: 'Brian',
+          }, function(err) {
+            err.message.should.equal('There are multiple instances found.' +
               'Upsert Operation will not be performed!');
-          done();
+            done();
+          });
         });
       });
-    });
+
+    bdd.itIf(connectorCapabilities.atomicUpsertWithWhere === true,
+      'upsertWithWhere update the first matching instance when multiple instances are ' +
+      'retrieved based on the filter criteria', async () => {
+        // The first matching instance is determinate from specific connector implementation
+        // For example for mongodb connector the sort parameter is used (default to _id asc)
```

Comment:
I find it problematic that connectors implement different behavior than our default non-atomic implementation. Are we sure that SQL connectors like MySQL and PostgreSQL will update the first record only, the same way as MongoDB connector works? It would be great if we could align juggler-provided non-atomic version with the behavior of real connectors (both SQL and noSQL). If different connectors behave differently, then IMO we need to clearly document these differences and probably need to add a new `connectorCapability` flag to allow the test suite to execute the right set of expectations?

Reply:
> Are we sure that SQL connectors like MySQL and PostgreSQL will update the first record only, the same way as MongoDB connector works?

@bajtos Currently only mongodb supports that, no other connectors have atomic implementation.

Hmm isn't `atomicUpsertWithWhere` already a connector flag? I will document the behaviour after PRs merge :)